parser/openfoodfacts_taxonomy_parser/unparser.py

```diff
@@ -0,0 +1,124 @@
+from neo4j import GraphDatabase
+
+
+class WriteFile:
+    def __init__(self, uri="bolt://localhost:7687"):
+        self.driver = GraphDatabase.driver(uri)
+        self.session = (
+            self.driver.session()
+        )  # Doesn't create error even if there is no active database
+
+    def normalized_filename(self, filename):
+        """add the .txt extension if it is missing in the filename"""
+        return filename + (
+            ".txt" if (len(filename) < 4 or filename[-4:] != ".txt") else ""
+        )
+
+    def get_all_nodes(self):
+        """query the database and yield each node with its parents,
+        this function use the relationships between nodes"""
+        query = """
+            MATCH path = ShortestPath( (h:TEXT{id:"__header__"})-[:is_before*]->(f:TEXT{id:"__footer__"}) )
+            UNWIND nodes(path) AS n
+            RETURN n , [(n)-[:is_child_of]->(m) | m ]
+        """
+        results = self.session.run(query)
+        for result in results:
+            node, parents = result.values()
+            yield node, parents
+
+    def list_tags_lc(self, node):
+        """return an ordered list of the language codes (lc) used in a node"""
+        lc_list = []
+        if "stopwords" in node["id"]:
+            # stopwords node only have a tags_lc property
+            key = "tags_"
+            part = 1
+        else:
+            key = "tags_ids_"
+            part = 2
+
+        for property in node:
+            if property.startswith(key):
+                lc_list.append(property.split("_", part)[part])
+        lc_list.sort()
+        return lc_list
+
+    def write_tags_line(self, node, lc):
+        """return a string that should look like the original line"""
+        line = (", ").join(node["tags_" + lc])
+        line = lc + ":" + line + "\n"
+        return line
+
+    def write_comment_lines(self, comments):
+        """return a string that should look like the original comment lines"""
+        lines = ("\n").join(comments)
+        lines += "\n"
+        return lines
+
+    def list_property_and_lc(self, node):
+        """return a ordered list of properties with their language code (lc)"""
+        prop_list = []
+        for property in node:
+            if property.startswith("prop_"):
+                prop_list.append(property[5:])
+
+        # there is no rule for the order of properties
+        # properties will be arranged in alphabetical order
+        prop_list.sort()
+        return prop_list
```

Comment:
typically done with a list comprehension:

```suggestion
        # there is no rule for the order of properties
        # properties will be arranged in alphabetical order
        return sorted(
            property[5:] for property in node if property.startswith("prop_")
        )
```